### PR TITLE
feat: soil ID algorithm service

### PR DESCRIPTION
**Pull request by shrouxm**
## Description
Needs techmatters/terraso-backend#1273 to be merged. Makes soil ID API calls available to clients, but doesn't try to figure out how to cache this in the redux store yet.

### Checklist
- [x] Corresponding issue has been opened

### Related Issues
Progress in techmatters/terraso-backend#1265 and techmatters/terraso-backend#1266

### CODE REVIEW

File: src/soilId/soilIdAlgorithmFragments.ts
Comment (shrouxm) [inline src/soilId/soilIdAlgorithmFragments.ts:1]: question: i feel like it'd be slightly nicer if these files were named `soilId{Fragments,Service}` and the ones that are named that now were named `soilData{Fragments,Service}` for consistency with backend. does anyone else care? if not then i can not bother with a mobile client PR to rewrite the imports

Comment (knipec) [reply src/soilId/soilIdAlgorithmFragments.ts:1]: I'd be in favor of this assuming it takes a reasonable amount of time :)

**Review by knipec — APPROVED** [review]

File: src/soilId/soilDataService.ts [hunk 1]
```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright © 2023 Technology Matters
```
Comment (paulschreiber) [inline src/soilId/soilDataService.ts:2]:
```suggestion
 * Copyright © 2024 Technology Matters
```

File: src/soilId/soilDataFragments.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright © 2023 Technology Matters
```
Comment (paulschreiber) [inline src/soilId/soilDataFragments.ts:2]:
```suggestion
 * Copyright © 2024 Technology Matters
```

**Pull request merged**